devops/scripts/deploy-public-upstream.sh

```diff
@@ -65,7 +65,24 @@ for commit in "${commits[@]}"; do
   fi
   echo "Adding $commit:"
   git --no-pager log --format=%B -n 1 "$commit"
-  git cherry-pick -rn -X theirs "$commit" 2>&1
+  if ! git cherry-pick -rn -X theirs -m 1 "$commit"; then
+    echo "Conflict detected in $commit. Checking for deleted files."
+    conflicted_files=$(git diff --name-only --diff-filter=U)
+    for file in $conflicted_files; do
+      if ! git ls-tree -r "$commit" --name-only | grep -q "^$file$"; then
+        echo "File $file was deleted in the cherry-picked commit. Resolving by keeping the deletion."
+        git rm "$file"
+      else
+        echo "Conflict required manual resolution for $file."
+      fi
+    done
+
+    # Stage the changes and continue the cherry-pick
+    git add -u
+    git commit --no-edit || {
+      echo "No changes to commit. Continuing."
+    }
```

Comment:
```suggestion
    git commit --no-edit || echo "No changes to commit. Continuing."
```
not familiar with what the brackets are doing here was that supposed to be a subshell `()`?

Reply:
¯\_(ツ)_/¯ it was a chatgpt suggestion. yours is easier to grok